app/models/manageiq/providers/kubernetes/container_manager_mixin.rb

```diff
@@ -124,31 +124,31 @@ def default_authentication_type
   end
 
   def scan_job_create(entity)
-    check_policy_prevent(:request_containerimage_scan, entity, :raw_scan_job_create, entity)
+    check_policy_prevent(:request_containerimage_scan, entity, :raw_scan_job_create)
   end
 
-  def raw_scan_job_create(entity)
+  def raw_scan_job_create(class_name, id)
     Job.create_job(
       "ManageIQ::Providers::Kubernetes::ContainerManager::Scanning::Job",
       :name            => "Container image analysis",
-      :target_class    => entity.class.name,
-      :target_id       => entity.id,
+      :target_class    => class_name,
+      :target_id       => id,
       :zone            => my_zone,
       :miq_server_host => MiqServer.my_server.hostname,
       :miq_server_guid => MiqServer.my_server.guid,
-      :ems_id          => id,
+      :ems_id          => class_name.constantize.find_by(:id => id).try(:ext_management_system),
```

Comment:
Leave it in. You are saying it is a separate bug. I get it.
Can you use a where and pluck? Will this object always belong to an ems/have the `ems_id` column?